src/app/qgisapp.h

```diff
@@ -1467,7 +1467,7 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
     void selectByForm();
 
     //! refresh map canvas
-    void refreshMapCanvas();
+    void refreshMapCanvas( bool redrawAllLayers = false );
```

Comment:
@nyalldawson , should the default value be true here? The refreshMapCanvas() is called from a bunch of places expecting the former behavior.

If we think it's safe to have a default set to false, you'll nevertheless need to change the mActionDraw connection:

`connect( mActionDraw, &QAction::triggered, this, &QgisApp::refreshMapCanvas );`

This is the action taking place when you hit f5 (or the refresh button in the toolbar) which we do want to keep refreshing all layers.

Reply:
Argh - I  missed the mActionDraw one, but I **did** check over all the other uses to make sure they only needed a single layer update.

Reply:
Fixed